partitura/score.py

```diff
@@ -3388,15 +3541,25 @@ def tie_notes(part):
                 note_id = _make_tied_note_id(cur_note.id)
             else:
                 note_id = None
-            next_note = Note(
-                note.step,
-                note.octave,
-                note.alter,
-                id=note_id,
-                voice=note.voice,
-                staff=note.staff,
-                symbolic_duration=sym_dur,
-            )
+            if isinstance(cur_note, UnpitchedNote):
+                next_note = UnpitchedNote(
+                    cur_note.step,
+                    cur_note.octave,
+                    id=note_id,
+                    voice=cur_note.voice,
+                    staff=cur_note.staff,
+                    symbolic_duration=sym_dur,
+                )
+            else:
+                next_note = Note(
+                    note.step,
+                    note.octave,
+                    note.alter,
+                    id=note_id,
+                    voice=note.voice,
+                    staff=note.staff,
+                    symbolic_duration=sym_dur,
+                )
```

Comment:
this is happening within a loop of ```for note in list(part.iter_all(Note)):``` and ```cur_note = note```, so ```isinstance(cur_note, UnpitchedNote)``` always evaluates to False as UnpitchedNotes and Notes are different classes. To include unpitched notes one could loop over  ```for note in list(part.iter_all(GenericNote, include_subclasses=True)):``` if grace notes are no problem. if unpitched notes are not important we can get rid of the redundancy.